compiler/optimizations/remoteValueForwarding.cpp

```diff
@@ -208,10 +209,107 @@ static void updateTaskFunctions(Map<Symbol*, Vec<SymExpr*>*>& defMap,
       // Would need to flatten them if they are not already.
       INT_ASSERT(isGlobal(fn));
 
+      // Map from an aggregate type to a list of its fields that will be RVF'd
+      std::map<Symbol*, std::vector<Symbol*> > fieldsToRVF;
+      typedef std::map<Symbol*, std::vector<Symbol*> >::iterator argIter;
+
       // For each reference arg that is safe to dereference
       for_formals(arg, fn) {
         if (canForwardValue(defMap, useMap, syncSet, fn, arg)) {
           updateTaskArg(useMap, fn, arg);
+          AggregateType* argType = toAggregateType(arg->type);
```

Comment:
Let's have a comment around here along the lines of "Identify reference fields that can be RVF'd in addition to the formal arg"